kew.js

```diff
@@ -407,8 +407,10 @@ Promise.prototype.done = function (onSuccess, onFailure) {
   var deferred = new Promise()
   var isTimeout = false
 
+  // Initialize the timeout here to capture better stack info
+  var timeoutErr = new Error(timeoutMsg || 'Promise timeout after ' + timeoutMs + ' ms.')
```

Comment:
most promise libraries consider this prohibitively expensive, and put it behind some sort of a dev flag (because instantiating an Error object has to materialize the current stack). See
https://github.com/kriskowal/q#long-stack-traces